presto-hive/src/test/java/com/facebook/presto/hive/TestHiveMaterializedViewLogicalPlanner.java

```diff
@@ -1077,6 +1089,551 @@ public void testBaseToViewConversionWithGroupBy()
+                    aggregation(
+                            singleGroupingSet("orderkey"),
+                            ImmutableMap.of(Optional.empty(), functionCall("sum", ImmutableList.of("sum_26"))),
+                            ImmutableList.of(),
+                            ImmutableMap.of(),
+                            Optional.empty(),
+                            FINAL,
+                            node(
+                                    ExchangeNode.class,
+                                    anyTree(
+                                            aggregation(
+                                                    ImmutableMap.of("sum_26", functionCall("sum", ImmutableList.of("count"))),
+                                                    PARTIAL,
+                                                    project(aggregation(
+                                                            ImmutableMap.of("count", functionCall("count", ImmutableList.of("count_27"))),
+                                                            FINAL,
+                                                            node(
+                                                                    ExchangeNode.class,
+                                                                    anyTree(
+                                                                            aggregation(
+                                                                                    singleGroupingSet("ds", "orderkey"),
+                                                                                    ImmutableMap.of(Optional.of("count_27"), functionCall("count", ImmutableList.of())),
+                                                                                    ImmutableList.of(),
+                                                                                    ImmutableMap.of(),
+                                                                                    Optional.empty(),
+                                                                                    PARTIAL,
+                                                                                    node(
+                                                                                            ProjectNode.class, constrainedTableScan(
+                                                                                                    table,
+                                                                                                    ImmutableMap.of("ds", multipleValues(createVarcharType(10), utf8Slices("2021-07-11", "2021-07-12"))),
+                                                                                                    ImmutableMap.of("orderkey", "orderkey", "ds", "ds")))))))))))));
+
+            //expect a query on the base table to behave as if the MV doesn't exist
+            PlanMatchPattern expectedPatternWithoutCountOptimization = anyTree(
```

Comment:
It validates that the base query does differ from the view query in the expected way when the data is completely stale. It seems at least marginally useful to me, but I'm not sure and would defer to you on this.

Reply:
We have other tests to verify what happens when optimization is not enabled

Reply:
@pranjalssh just to be sure we're clear- the `PlanMatchPattern` you highlighted, `expectedPatternWithoutCountOptimization`, _is_ tested with the mv optimization config enabled. The name is admittedly confusing, I chose it because we're expecting the optimization to not occur (despite being ran with the setting on). The `PlanMatchPattern` that is tested _without_ the optimization enabled is the view query. That's the one you had in mind to remove, right?

Reply:
Yeah I meant the other one